src/lib/support/ReturnMacros.h

```diff
@@ -0,0 +1,62 @@
+/*
+ *
+ *    Copyright (c) 2020 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#pragma once
+
+#include <core/CHIPError.h>
+
+/// Returns the error code if the expression returns something different
+/// than CHIP_NO_ERROR
+///
+/// Use like:
+///   ReturnErrorOnFailure(channel->SendMsg(mnsg));
+#define ReturnErrorOnFailure(expr)                                                                                                 \
+    {                                                                                                                              \
+        CHIP_ERROR __err = (expr);                                                                                                 \
+        if (__err != CHIP_NO_ERROR)                                                                                                \
+        {                                                                                                                          \
+            return __err;                                                                                                          \
+        }                                                                                                                          \
+    }                                                                                                                              \
+    (void) 0
```

Comment:
I would strongly prefer we use:
```
do {
  ...
} while (0)
```
so the whole thing is in fact the single statement it looks like it is.  That way if people use it in `if ... else` without braces (which ideally they wouldn't do, I know), it will work correctly instead of giving confusing compiler errors.

Similar for the other macros defined here.